src/desktop/lib/logoutHandler.ts

```diff
@@ -0,0 +1,29 @@
+import type { LogoutEventOptions } from "typings/mediator"
+
+// const FlashMessage = require("../components/flash/index.coffee")
+
+export const logoutEventHandler = (options?: LogoutEventOptions) => {
+  const redirectPath = options?.redirectPath
+
+  fetch("/users/sign_out", {
+    headers: {
+      // TODO: To trick the logout page into thinking we made and xhr request
+      "X-Requested-With": "XMLHttpRequest",
+    },
+    method: "DELETE",
+  })
+    .then(() => {
+      if (window.analytics && window.analytics.reset) {
+        // guard for aggressive ad blockers
+        window.analytics.reset()
+      }
+      if (redirectPath) {
+        location.assign(redirectPath)
+      } else {
+        location.reload()
+      }
+    })
+    .catch(errorMessage => {
+      // new FlashMessage({ message: errorMessage })
```

Comment:
We can use this error modal to display the message: https://github.com/artsy/force/blob/be156e9feda1505b3a231e6c3ea347d1a1a02cb9/src%2Fv2%2FComponents%2FModal%2FErrorModal.tsx#L18